core/kazoo_number_manager/src/knm_phone_number.erl

```diff
@@ -326,6 +326,10 @@ save(T0=#{todo := PNs, options := Options}) ->
 save(PN=#knm_phone_number{is_dirty = false}) ->
     lager:debug("not dirty, skip saving ~s", [number(PN)]),
     PN;
+save(PN=#knm_phone_number{state=?NUMBER_STATE_AVAILABLE
+                         ,assigned_to=AssignedTo
+                         }) when AssignedTo =/= 'undefined' ->
+    save(PN#knm_phone_number{assigned_to='undefined'});
```

Comment:
ehhh this should happen in unwind/disconnect or knm_number_states instead of here.
Also, you can find existing tests and for every number getting into the available state check for this field's value.
I would especially be interested in a test about the release of a non local number with reserve history.